### DISCUSSION TITLE
Alembic Primary Key issue with mssql ( Azure Synapse SQL DW )

### DISCUSSION THREAD

**Discussion by ashishmgofficial** [sqlalchemy/alembic#939] [Usage Questions] [answered]
**Describe the bug**
Im trying to apply migrations using alembic to Azure Synapse SQL DW. Im facing following issue while performing alembic upgrade head:
```
sqlalchemy.exc.ProgrammingError: (pyodbc.ProgrammingError) ('42000', '[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Enforced unique constraints are not supported. To create an unenforced unique constraint you must include the NOT ENFORCED syntax as part of your statement. (104467) (SQLExecDirectW)')
[SQL: 
CREATE TABLE alembic_version (
        version_num VARCHAR(32) NOT NULL, 
        CONSTRAINT alembic_version_pkc PRIMARY KEY (version_num)
)

]
(Background on this error at: https://sqlalche.me/e/14/f405)
```

**Expected behavior**
Version table created normally and migration successful

**To Reproduce**
my `alembic.ini`  configuration for mssql:
```
sqlalchemy.url = mssql+pyodbc://{user}:{password}@{host/server}:1433/{db}?autocommit=True&driver=ODBC+Driver+17+for+SQL+Server

```

**Error**

```
sqlalchemy.exc.ProgrammingError: (pyodbc.ProgrammingError) ('42000', '[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Enforced unique constraints are not supported. To create an unenforced unique constraint you must include the NOT ENFORCED syntax as part of your statement. (104467) (SQLExecDirectW)')
[SQL: 
CREATE TABLE alembic_version (
        version_num VARCHAR(32) NOT NULL, 
        CONSTRAINT alembic_version_pkc PRIMARY KEY (version_num)
)

]
(Background on this error at: https://sqlalche.me/e/14/f405)
```

**Versions.**
 - OS: Linux Ubuntu 20.04
 - Python: 3.9.7
 - Alembic: 1.7.3
 - SQLAlchemy: 1.4.25
 - Database: Azure Synapse SQL DW
 - DBAPI:

**Additional context**

**Have a nice day!**

**Comment by ashishmgofficial** [comment]
I tried adding the config `version_table_pk = false` as suggested in the comments in this SO issue I raised  [here](https://stackoverflow.com/questions/69367506/alembic-primary-key-issue-with-mssql-azure-synapse-sql-dw?noredirect=1#comment122607182_69367506) . But it doesnt seem to have any effect

**Comment by gordthompson** [comment, marked as answer]
I was able to suppress the creation of the primary key on the alembic_version table by doing the following:

Edit env.py and add the `version_table_pk=False` option to the `context.configure()` call in the `run_migrations_online()` function. That is, this …

```python
    with connectable.connect() as connection:
        context.configure(
            connection=connection, target_metadata=target_metadata
        )
```

… becomes this

```python
    with connectable.connect() as connection:
        context.configure(
            connection=connection, target_metadata=target_metadata,
            version_table_pk=False
        )
```

**Reply by CaselIT** [reply]
it's documented as such here https://alembic.sqlalchemy.org/en/latest/api/runtime.html?#alembic.runtime.environment.EnvironmentContext.configure.params.version_table_pk
